docs/src/architecture/domain-model.md

```diff
@@ -0,0 +1,217 @@
+    * lists available log files
+    * retrieve content of individual log files
+  * **Device Monitoring**, collects metrics from the device and forward these to the IoT cloud
+    * allows monitors the health of devices
+    * helps to troubleshoot when problems with the device are reported
+
+## thin-edge device concept
+
+**thin-edge** facilitates IoT functionality to the device it is running on, as well as to devices that are connected to that device.
+
+  * the device thin-edge is running on is referred as the **main-device**
+    * thin-edge on the **main-device** establishes and manages all communication to the cloud
+  * all devices connected to the **main-device** are referred as **external child-devices**
+  * each device is represented in the cloud with an individual **device twin**
+  * a unique **child-id** makes the association between each **external child-device** and it's **device twin**
+  * all telemetry data (i.E. Measurements, Variables, Events, Alarms) of a device (i.E. **main-device** or **external child-devices**) appear
+    in the cloud in context of its corresponding **device twin**
+
+The figure below illustrates the device concept.
+
+![Device Concept](images/device-concept.svg)
+
+## thin-edge data concept
+
+**thin-edge** provides APIs to easily connect device's **data points** to cloud's **telemetry data handling**.
+Usually there are two kinds of SW components that use those **thin-edge** APIs:
+  * the **domain application**, to provide **data points** from it's processing flow and potentially inputs/outputs from _sensors/actuators_ to **thin-edge**
+  * **protocol drivers**, to
+    * provide purely inputs/outputs from _sensors/actuators_, e.g. from protocols as OPC UA, Profinet, CANopen, ...
+    * access the **domain application** (e.g. when the **domain application** cannot or shall not access **thin-edge** APIs directly) to provide **domain application's** **data points** to **thin-edge**
+
+Per **telemetry data** the interpretation of **data points** differs:
+
+  * **Measurements**:
+    * a **measurement** is represented with one or more **data points**
+      and a reference to the corresponding **metric**
+    * the values of the **data points** reflect the measurement values
+    * **thin-edge** puts the measurement into the context of the corresponding metric and sends it to the cloud
+  * **Command**:
+    * a **command** is represented with one or more values received from the **cloud**
+    * **thin-edge** provides those values to the **domain application** and **protocol drivers** as **data point** values
+  * **Events**:
+    * for events **thin-edge** does not get in touch with **data points** at all;<br/>
+      instead any event related **data point** and it's value is evaluated by the event producer internally (i.E. the **domain application** or **protocol drivers**)
+    * the event producer sends a coresponding notification to **thin-edge** whenever an event shall be raised
```

Comment:
I really don't understand what translation is done - if any, by who and how.

Reply:
I added and improved some details about events, see https://github.com/thin-edge/thin-edge.io/commit/34f0bc731df6a79bab500fda06dae6ab93a60a3d

Does that solve the confusion?